mcs/class/System.Web/System.Web/HttpRequest.cs

```diff
@@ -2200,7 +2200,7 @@ long MoveToNextBoundary ()
 
 					if (buffer [bl - 2] == '-' && buffer [bl - 1] == '-') {
 						at_eof = true;
-					} else if (buffer [bl - 2] != CR || buffer [bl - 1] != LF) {
+					} else if (!(buffer[bl - 2] == CR && buffer[bl - 1] == LF) && buffer[bl - 2] != LF) {  //CRLF or LF
```

Comment:
This does not look right. You should always check the last character in the stream, so `- 2` is not doing that and it won't be correct if LF is the last character of the stream.

Reply:
Well this is not exactly the stream this is just a buffer that is the size of the boundary +2.  The two additional characters are added to reserve space for either CRLF or "--".  The buffer only gets written to once two "--" characters are found at the beginning of a new line (state = 1).  The CompareBytes then validates if the buffer matches the boundary string (excluding the last 2 characters).

If there is no CR in the line ending of the boundary the last character of the buffer is actually the **first character of the next line**.  Thus the reason bl-2 and bl-1 are evaluated for LF.

The .Net framework implementation does this a bit differently as it reads new lines rather than individual characters when its searching for the next boundary.  I could update this implementation to follow that pattern, but its going to be a much larger lift.